libs/datasets/dataset_utils.py

```diff
@@ -13,6 +17,42 @@
 
 _logger = logging.getLogger(__name__)
 
+class DataVersion(object):
```

Comment:
Do you think you could move the data version related code out into it's own file? 

dataset_utils is already a bit overloaded and I think the data version code has enough functionality that it merits its own file!

Reply:
Done.